app/scripts/metamask-controller.js

```diff
@@ -492,11 +493,20 @@ export default class MetamaskController extends EventEmitter {
     this.selectedNetworkController = new SelectedNetworkController({
       messenger: this.controllerMessenger.getRestricted({
         name: 'SelectedNetworkController',
-        allowedActions: ['NetworkController:getNetworkClientById'],
+        allowedActions: [
+          'NetworkController:getNetworkClientById',
+          'NetworkController:getState',
+          `PermissionController:hasPermissions`,
+        ],
         allowedEvents: ['NetworkController:stateChange'],
       }),
+      state: initState.SelectedNetworkController,
     });
 
+    const networkClient = this.networkController.getProviderAndBlockTracker();
+    this.provider = networkClient.provider;
+    this.blockTracker = networkClient.blockTracker;
```

Comment:
Following 9af40f307ef4dcb2536aee62a5aef6ebb3126f6e, this seems to no longer be used

```suggestion
```

Reply:
removed here: https://github.com/MetaMask/metamask-extension/pull/23130/commits/bf5bc3ed7b7f769934f626e0752342261f87ba70